data/ledger.go

```diff
@@ -226,6 +226,9 @@ func (l *Ledger) BalanceAndStatus(addr basics.Address) (money basics.MicroAlgos,
 	}
 	moneyWithoutPendingRewards = dataWithoutRewards.MicroAlgos
 
+	totalCurrency = dataWithoutRewards.ThisCurrencyTotal
+	currencies = dataWithoutRewards.Currencies
```

Comment:
You're returning a map ( i.e. a pointer ); it's likely to be ok in most cases, as long as we use it for read-only.
If this is not the case, we'll need to either copy that or to lock that.

Reply:
Sure, might as well be safe and clone the map.

The intent was that the caller is read-only, which is true of current callers, and elsewhere in the code we explicitly copy the map when modifying -- like clone() in data/transactions/currency.go -- but seems like a reasonable idea to guard against buggy callers here.

Reply:
thanks!